crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -556,7 +557,10 @@ async fn send_archived_segment_notification(
 
     archived_segment_notification_sender.notify(move || archived_segment_notification);
 
-    while acknowledgement_receiver.next().await.is_some() {
-        // Wait until all acknowledgements are received
+    if acknowledgement_receiver.next().await.is_some() {
```

Comment:
Why was this changed to `if`? There could be multiple acknowledgements and we need to wait for all of them, this is why we had `while` and corresponding comment.